Glitter chemical recipes

theoctopusempress opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request
-Adds recipes for glitter to chemistry reagents as well as for all colours of glitter (pink, white, and blue) 
-Changed name of Generic glitter to standard glitter and changed description to something more relevant
-Recipe for Standard glitter is 5u oil, 2u sulphuric acid, 3u ash, 2u aluminum, and heating to 340k since real life glitter is made by adding aluminum lining to plastic dust
- recipes for pink, white, and blue involve adding 5u generic glitter and 1u of stable plasma, sugar, or nitrogen respectively
<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game
Glitter as it currently exists in the game can only be found in grenades as a potential arcade machine prize, making it rare to find in regular play. It has plenty of potentially mischief for role-play centered roles such as magicians, clowns and mimes and can be used for pranks as pink glitter looks identical to plasma floods. As an added intended effect, the recipe for generic glitter is similar to that for plastic sheets and its ratios are ideal for the X-large beakers that require plastic sheets. This encourages chemists to produce plastic sheets early in the round, ensuring that chemists who wish to play with toys first complete their duties.
<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding.
-->

## Testing Photographs and Procedure
<!-- Include any screenshots/videos/debugging steps of the modified code functioning successfully, ideally including edge cases. -->
<details>
<summary>Screenshots&Videos</summary>

![glitterbar](https://github.com/BeeStation/BeeStation-Hornet/assets/78881224/3dcecd27-7418-4b33-a88b-c42f9dff1d21)



https://github.com/BeeStation/BeeStation-Hornet/assets/78881224/f97c638a-5b19-4b06-81e0-143bfe7f4690




</details>

## Changelog
:cl:
add: Recipe for Standard glitter
add: Recipes for pink, white and blue glitter
tweak: Changed name of Generic glitter to Standard glitter and changed its description
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

theoctopusempress committed:
Glitter chemical recipes

-Adds recipes for glitter to chemistry reagents as well as for all colours of glitter (pink, white, and blue)
-Changed name of Generic glitter to standard glitter and changed description to something more relevant

EvilDragonfiend commented on code/modules/reagents/chemistry/recipes/others.dm:
```diff
@@ -726,6 +726,31 @@
+
+/datum/chemical_reaction/glitter/pink
+	name = "glitter"
+	id = /datum/reagent/glitter
```
wrong copy pasta.
name and id should be unique.
do the same to other reactions.

EvilDragonfiend commented on code/modules/reagents/chemistry/recipes/others.dm:
```diff
@@ -726,6 +726,31 @@
+	name = "glitter"
+	id = /datum/reagent/glitter
+	results = list(/datum/reagent/glitter/blue = 5)
+	required_reagents = list(/datum/reagent/glitter = 5, /datum/reagent/nitrogen =1)
```
minor nitpick. =1 needs a whitespace.

theoctopusempress committed:
Gave each recipe a unique name and ID

Changed recipes names and ID to be unique to each recipe and added relevant whitespace to blue glitter recipe as per comment by EvilDragonfiend.

EvilDragonfiend commented on code/modules/reagents/chemistry/recipes/others.dm:
```diff
@@ -726,6 +726,31 @@
 
+/datum/chemical_reaction/glitter
+	name = "standard glitter"
+	id = /datum/reagent/glitter/standard
```
There's no such reagent.

EvilDragonfiend replied:
You didn't do anything to this yet.

theoctopusempress committed:
Changed "standard glitter" reagent id to align with recipe ID

Changed /datum/reagent/glitter to /datum/reagent/glitter/standard.

EvilDragonfiend commented on code/modules/reagents/chemistry/reagents/other_reagents.dm:
```diff
@@ -2010,9 +2010,9 @@
-/datum/reagent/glitter
-	name = "Generic Glitter"
-	description = "If you can see this description, contact a coder."
+/datum/reagent/glitter/standard
```
```suggestion
/datum/reagent/glitter
```
What you made is actually making it not-standard.
/datum/reagent/glitter has no definition now.

theoctopusempress committed:
Update code/modules/reagents/chemistry/reagents/other_reagents.dm

Co-authored-by: EvilDragonfiend <87972842+EvilDragonfiend@users.noreply.github.com>

theoctopusempress committed:
Changed ID in recipe name to the correct one

id = /datum/reagent/glitter/standard changed to id = /datum/reagent/glitter

EvilDragonfiend commented on code/modules/reagents/chemistry/reagents/other_reagents.dm:
```diff
@@ -2011,8 +2011,8 @@
-	name = "Generic Glitter"
-	description = "If you can see this description, contact a coder."
+	name = "standard glitter"
+	description = "Standard sparkles that get everywhere."
```
Name and description should be "white glitter" thingies rather than saying standard. It's something standard in the code, but it's not contextual ingame.

theoctopusempress replied:
There is currently also white glitter on line 2033, and I think that this one looks more like light pink anyways.

theoctopusempress committed:
Changed "standard glitter" to "light pink glitter"

EvilDragonfiend commented on code/modules/reagents/chemistry/reagents/other_reagents.dm:
```diff
@@ -2011,8 +2011,8 @@
-	description = "If you can see this description, contact a coder."
+	name = "light pink glitter"
+	description = "light pink sparkles that get everywhere."
 	color = "#FFFFFF" //pure white
```
the comment confuses me. I think it should say the base color of the decal is light pink

theoctopusempress replied:
It confused me too, its a holdover from before I started working on it. I think its applying a white filter onto a sprite that was darker pink/purple to end up with a light pink sprite. I will tinker with it this evening to figure out what is going on.

theoctopusempress replied:
After further tests, it seems that this line doesn't affect the colour but removing it stops a number of recipes from working so I am changing the comment and leaving it alone.

theoctopusempress committed:
Changed comment regarding base colour of decal

changed from "pure white" to "base colour of decal is light pink"
the colour seems to have no affect on the decal's visuals, however removing it  caused a number of recipes to stop working.

EvilDragonfiend reviewed (approved):
All good. Change wiki chem page then ping me. I will merge it after you edit it.
